Issue 6

bryce-seefieldt opened the pull request:
Hey, I tried my best to work within your requested preferences however:
1. I didnt make src/builder/markdown_parser.py /generate_markdown function  as the added formatting was a simple addition to the html_builder.py fucntionality, perhaps you will find a better way to approach this, perhaps using the line queries module, I wasn't able to figure out how to utilize it properly and incorporate the my regEx .MD reformat approach.
Perhaps you can actually teach me a bit more about the .Doc module and it's funtionality to take a better approach to the incorporating .MD formating.

## Issue 6 Changes Made:
### ./src/til-builder_main.py
Line 43 - Added "OR" condition for .md file verification
Line 45 - Updated error message to include markdown

### ./builder/html_builder.py
Line 3 Imported re (regular expressions) module
Lines 85 to 89 - Added regEx function to convert Markdown bold and italic formatting to html tags

### ./models/file.py 9-11
Lines 9 to 11 - IF statement to create .html file from .md extension

#### ReadMe.md
Line 34

humphd commented on src/builder/html_builder.py:
```diff
@@ -79,7 +81,12 @@ def generate_html_for_file(file_path):
                                 text(paragraph_content.replace(line_queries.H6_TOKEN, ""))
                         else: 
-                            with tag('p'):
+                            with tag('p'):   
```
@bryce-seefieldt how are the changes coming?

bryce-seefieldt commented:
Not great but I’m working to complete them by this afternoon at the latest.
I’ve communicated that to Amnish (author).

My apologies for the delay.

Bryce

On Tue, Sep 26, 2023 at 10:18 AM David Humphrey ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/builder/html_builder.py
> <https://github.com/Amnish04/til-page-builder/pull/7#discussion_r1337290555>
> :
>
> > @@ -79,7 +81,12 @@ def generate_html_for_file(file_path):
>                              with tag('h6'):
>                                  text(paragraph_content.replace(line_queries.H6_TOKEN, ""))
>                          else:
> -                            with tag('p'):
> +                            with tag('p'):
>
> @bryce-seefieldt <https://github.com/bryce-seefieldt> how are the changes
> coming?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Amnish04/til-page-builder/pull/7#discussion_r1337290555>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6JISL7CQO2NRLD2HW6OIVDX4LP3ZANCNFSM6AAAAAA5EKYB4M>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

bryce-seefieldt commented:
Hi, I updated the html_builder.py with your suggestion. Was not able to find a streamlined solution for the file.py conditions to identify the correct file extension, perhaps we can discuss further.  I also deleted additional test files that were in the wrong location.  Please let me know if the changes made resolve the issue.

Amnish04 commented:
Hi.
For the file.py issue, what I mean is just replace the hardcoded conditional branches
![image](https://github.com/Amnish04/til-page-builder/assets/78865303/5dbdc6d5-8cb6-41e6-962e-543465f63002)

with the following statement.
```
write_path = self.file_path.replace(os.path.splitext(self.file_path), '.html')
```

This takes care of any new formats we might add in the future.

Also, where you replaced `text` with `asis` function
```
with tag('p'):   
                                if has_md_extension(file_path):   
                                    # Use regEx to replace markdown bold format (** **) with html <strong> </strong> tags                              
                                    paragraph_content = re.sub(r'\*\*(.*?)\*\*', r'<strong>\1</strong>', paragraph_content)
                                    # Use regEx to replace markdown italic format (* * or _ _) with html <em> </em> tags                              
                                    paragraph_content = re.sub(r'(\*|_)(.*?)\1', r'<em>\2</em>', paragraph_content)
                                doc.asis(paragraph_content)
```

Could you please make this minor change, that only calls `asis` in case of `.md` files. In other cases, it should call normal `text` function because we should be escaping special characters in that case.

Here's what it should look like:
```
with tag('p'):   
                                if has_md_extension(file_path):   
                                    # Use regEx to replace markdown bold format (** **) with html <strong> </strong> tags                              
                                    paragraph_content = re.sub(r'\*\*(.*?)\*\*', r'<strong>\1</strong>', paragraph_content)
                                    # Use regEx to replace markdown italic format (* * or _ _) with html <em> </em> tags                              
                                    paragraph_content = re.sub(r'(\*|_)(.*?)\1', r'<em>\2</em>', paragraph_content)
    doc.asis(paragraph_content)
else:
    text(paragraph_content)
```

bryce-seefieldt commented:
As requested, issues fixed in:
html_builder.py -
Line 84-92; Added conditional statements for multiple file types 

file.py - Was able to implement streamlined code for replacing extension for write_path; 
Line 10; had to add the index ([1]) of .splitext return value, as well as passing that value as a string to .replace function:

`write_path = self.file_path.replace(f'{os.path.splitext(self.file_path)[1]}', '.html')`

![image](https://github.com/Amnish04/til-page-builder/assets/127043887/de992737-0dfc-44ce-a025-3d91082e09c4)
(as well, I imported 'os', Line 1)

Amnish04 commented:
Just reviewed the changes, eveything looks good now.
Thanks a lot for your cooperation and all the hard work.
This fixes #7, merging the PR now!
